packages/beacon-node/src/chain/chain.ts

```diff
@@ -1195,12 +1195,13 @@ export class BeaconChain implements IBeaconChain {
         const validatorIndices = this.beaconProposerCache.getValidatorIndices();
         const targetCustodyGroupCount = getValidatorsCustodyRequirement(headState, validatorIndices, this.config);
         this.custodyConfig.updateTargetCustodyGroupCount(targetCustodyGroupCount);
+        this.logger.verbose(`Updated targetCustodyGroupCount=${this.custodyConfig.targetCustodyGroupCount}`);
```

Comment:
if `targetCustodyGroupCount` is unchanged, we even don't need to emit `ChainEvent.updateTargetGroupCount`

Reply:
Updated to only perform updates and emit events if `targetCustodyGroupCount` as changed: https://github.com/ChainSafe/lodestar/pull/7736/commits/0b07d416b9a650c126afd4af192e7294462f5dc5